src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,101 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static int DEFAULT_CAPACITY = 16;
+    private static double LOAD_FACTOR = 0.75;
+    private int size = 0;
+    private Node<K,V>[] bucket;
+    private int threshold = 0;
+
+    public MyHashMap() {
+        this.bucket = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resizeIfNedeed();
+        int hash = getHash(key);
+        int index = hash % bucket.length;
```

Comment:
lets implement method getIndex to directly access index without calculation
```suggestion
        int index = getIndex(key);
```